Mathlib/MeasureTheory/Measure/Haar/Unique.lean

```diff
@@ -207,13 +228,12 @@ lemma integral_isMulLeftInvariant_isMulRightInvariant_combo
       conv_rhs => rw [← integral_mul_right_eq_self _ x]
   _ = (∫ y, f y * (D y)⁻¹ ∂ν) * ∫ x, g x ∂μ := integral_mul_left _ _
 
-/-- **Uniqueness of left-invariant measures**: Given two left-invariant measures which are finite on
+/-- Given two left-invariant measures which are finite on
 compacts, they coincide in the following sense: they give the same value to the integral of
 continuous compactly supported functions, up to a multiplicative constant. -/
 @[to_additive exists_integral_isAddLeftInvariant_eq_smul_of_hasCompactSupport]
-lemma exists_integral_isMulLeftInvariant_eq_smul_of_hasCompactSupport
-    (μ' μ : Measure G) [IsFiniteMeasureOnCompacts μ] [IsFiniteMeasureOnCompacts μ']
-    [IsMulLeftInvariant μ] [IsMulLeftInvariant μ'] [IsOpenPosMeasure μ] :
+lemma exists_integral_isMulLeftInvariant_eq_smul_of_hasCompactSupport (μ' μ : Measure G)
```

Comment:
There is something bizarre here (for me, at least) that mu and mu' have different kind of hypotheses.
Can one compare two left invariant measures without knowing that one of them is positive on open subsets and the other one is finite on compacts?

Reply:
If mu is the zero measure, you can not write mu' = c mu for some c. On the other hand, it's not a problem if mu' is the zero measure. That's why there is an assumption that mu is positive on open sets while there is no such assumption for mu', here and below.

Reply:
I have expanded the docstring of `haarScalarFactor` to insist on this point.